### ISSUE TITLE
empty array: $downloadOptions->getAllFormats()

### ISSUE THREAD

**Issue by alil0rd** [Athlon1600/youtube-downloader#190]
Hi,
First of all thanks for your great efforts to provide us this.
Recently, I have faced an issue.
I have got an empty response in all my requests!
 "$downloadOptions->getAllFormats()" returns an empty array.

I attached the  "$youtube -> getDownloadLinks($url)" response.

[225172991youtube.txt](https://github.com/Athlon1600/youtube-downloader/files/14941896/225172991youtube.txt)

**Comment by amiyoghoshstar** [comment]
I am also having the same issue.

**Comment by NumairNahiyo** [comment]
yes exactly i am also having the same issue

**Comment by CodeRevolutionPlugins** [comment]
Same issue

**Comment by ghost** [comment]
Same issue, how to solve it ?

**Comment by HafizAmeerHamza** [comment]
**Hello everyone,** 
I got the same issue and then I started tracking why it stopped working.
The issue arise due to the **client configurations.**  That's why the `$downloadOptions->getAllFormats()` returns an empty array.
I have been working with this and have tracked the issue. Will share and update once its all fixed.

**Comment by anaseem80** [comment]
> **Hello everyone,** I got the same issue and then I started tracking why it stopped working. The issue arise due to the **client configurations.** That's why the `$downloadOptions->getAllFormats()` returns an empty array. I have been working with this and have tracked the issue. Will share and update once its all fixed.

Solved?

**Comment by HafizAmeerHamza** [comment]
> > **Hello everyone,** I got the same issue and then I started tracking why it stopped working. The issue arise due to the **client configurations.** That's why the `$downloadOptions->getAllFormats()` returns an empty array. I have been working with this and have tracked the issue. Will share and update once its all fixed.
> 
> Solved?

Yes. Please update the client details with these in your **YoutubeDownloader.php** file. It will work.

```
"client" : {
            "hl" : "en",
            "clientName" : "WEB",
            "clientVersion" : "2.20210721.00.00",
            "clientFormFactor" : "UNKNOWN_FORM_FACTOR",
            "clientScreen" : "WATCH",
            "mainAppWebInfo" : {
                "graftUrl" : "/watch?v=JIbfzmt2LuY"
            }
        }
```
I will also share an updated version of this package in my repo. Keep checking.
